src/src/components/ObjectBrowser.jsx

```diff
@@ -3007,6 +3007,8 @@ class ObjectBrowser extends Component {
                     } else {
                         delete this.objects[event.id];
                     }
+                } else if (event.obj) {
```

Comment:
I personally would prefer, remove line 3004 and put in 3007 `else if (this.objects[event.id])` 

From my perspective this would unwire the code a bit

Reply:
Should be resolved by new approach.
Previously I tried to keep the change minimal to reduce workload for review, but of course the new variant is cleaner